src/helpers/subscribe.ts

```diff
@@ -0,0 +1,208 @@
+// TODO: need to see if EventEmitter works in the browser
+import { EventEmitter } from "node:events";
```

Comment:
There are browser based alternatives if not. There might even be some that work in both.